File: backend-csharp/Models/SanPham.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class SanPham
{
    public long Id { get; set; }

    public string TenSanPham { get; set; } = null!;

    public string Slug { get; set; } = null!;

    public long DanhMucId { get; set; }

    public long NhaCungCapId { get; set; }

    public string? MoTaNgan { get; set; }

    public string? MoTaDayDu { get; set; }

    public string? ThuongHieu { get; set; }

    public string TrangThai { get; set; } = null!;

    public bool NoiBat { get; set; }

    public int LuotXem { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public int LuotMua { get; set; }

    public int? CanNang { get; set; }

    public int? ChieuDai { get; set; }

    public int? ChieuRong { get; set; }

    public int? ChieuCao { get; set; }

    public string? MetaTitle { get; set; }

    public string? MetaDescription { get; set; }

    public virtual ICollection<BienTheSanPham> BienTheSanPhams { get; set; } = new List<BienTheSanPham>();

    public virtual ICollection<DanhGiaSanPham> DanhGiaSanPhams { get; set; } = new List<DanhGiaSanPham>();

    public virtual DanhMuc DanhMuc { get; set; } = null!;

    public virtual ICollection<HinhAnhSanPham> HinhAnhSanPhams { get; set; } = new List<HinhAnhSanPham>();

    public virtual NhaCungCap NhaCungCap { get; set; } = null!;

    public virtual ICollection<ThuocTinhSanPham> ThuocTinhSanPhams { get; set; } = new List<ThuocTinhSanPham>();

    public virtual ICollection<YeuThich> YeuThiches { get; set; } = new List<YeuThich>();
}

File: backend/models/SanPham.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const SanPham = sequelize.define(
  "SanPham",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    ten_san_pham: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    slug: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    danh_muc_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    nha_cung_cap_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    mo_ta_ngan: {
      type: DataTypes.STRING(500),
      allowNull: true,
    },
    mo_ta_day_du: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    thuong_hieu: {
      type: DataTypes.STRING(100),
      allowNull: true,
    },
    trang_thai: {
      type: DataTypes.STRING(20),
      allowNull: false,
    },
    noi_bat: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
    },
    luot_xem: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0,
    },
    can_nang: {
      type: DataTypes.INTEGER,
      allowNull: true,
    },
    chieu_dai: {
      type: DataTypes.INTEGER,
      allowNull: true,
    },
    chieu_rong: {
      type: DataTypes.INTEGER,
      allowNull: true,
    },
    chieu_cao: {
      type: DataTypes.INTEGER,
      allowNull: true,
    },
    meta_title: {
      type: DataTypes.STRING(255),
      allowNull: true,
    },
    meta_description: {
      type: DataTypes.STRING(500),
      allowNull: true,
    },
  },
  {
    tableName: "SanPham",
    timestamps: true,
    createdAt: "created_at",
    updatedAt: "updated_at",
  },
);

const BienTheSanPham = require("./BienTheSanPham");
const ThuocTinhSanPham = require("./ThuocTinhSanPham");
const HinhAnhSanPham = require("./HinhAnhSanPham");
const DanhGiaSanPham = require("./DanhGiaSanPham");

SanPham.hasMany(BienTheSanPham, { foreignKey: "san_pham_id", as: "bien_the" });
BienTheSanPham.belongsTo(SanPham, { foreignKey: "san_pham_id" });

SanPham.hasMany(ThuocTinhSanPham, {
  foreignKey: "san_pham_id",
  as: "thuoc_tinh",
});
ThuocTinhSanPham.belongsTo(SanPham, { foreignKey: "san_pham_id" });

SanPham.hasMany(HinhAnhSanPham, { foreignKey: "san_pham_id", as: "hinh_anh" });
HinhAnhSanPham.belongsTo(SanPham, { foreignKey: "san_pham_id" });

SanPham.hasMany(DanhGiaSanPham, { foreignKey: "san_pham_id", as: "danh_gia" });
DanhGiaSanPham.belongsTo(SanPham, { foreignKey: "san_pham_id" });

module.exports = SanPham;
